clients/privacy-center/pages/_app.tsx

```diff
@@ -17,23 +17,30 @@ import "@fontsource/inter/700.css";
 import store, { persistor } from "~/app/store";
 import "~/config/config.css";
 import theme from "~/theme";
+import { useFidesUserDeviceIdCookie } from "~/common/hooks/useCookie";
 
 const SafeHydrate: React.FC = ({ children }) => (
   <div suppressHydrationWarning>
     {typeof window === "undefined" ? null : children}
   </div>
 );
 
-const MyApp = ({ Component, pageProps }: AppProps) => (
-  <SafeHydrate>
-    <Provider store={store}>
-      <PersistGate loading={null} persistor={persistor}>
-        <ChakraProvider theme={theme}>
-          <Component {...pageProps} />
-        </ChakraProvider>
-      </PersistGate>
-    </Provider>
-  </SafeHydrate>
-);
+const MyApp = ({ Component, pageProps }: AppProps) => {
+  // Ensure each privacy centre has a unique fides_user_device_id
+  // for recording anonymised consent request preferences
+  useFidesUserDeviceIdCookie();
```

Comment:
since `useFidesUserDeviceIdCookie` automatically sets if it doesn't exist already, I'm thinking maybe we don't actually need to call it here—as long as everyone who needs the cookie uses this func, then we will always have a uuid, right? i.e. we could only call this in `ConsentRequestForm.tsx` instead, and we'd still have the same functionality?

(I know I was the one who suggested putting it in `_app.tsx` in the first place, but now I'm second guessing 😄 )

Reply:
moved it into the `onSubmit` of `ConsentRequestForm`